charts,salt,build: Bump Prometheus Adapter chart to 3.0.2

TeddyAndrieux opened the pull request:
Bump the Prometheus Adapter chart to 3.0.2 and also bump the
prometheus-adapter image to v0.9.1.

NOTE: The label selector for the Deployment changed we need to
remove the deployment before doing the upgrade or the downgrade.

TeddyAndrieux committed:
charts,salt,build: Bump Prometheus Adapter chart to 3.0.2

Bump the Prometheus Adapter chart to 3.0.2 and also bump the
prometheus-adapter image to v0.9.1.

Upgrade Prometheus Adapter chart using:
```
rm -rf charts/prometheus-adapter/
helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
helm repo update
helm fetch -d charts --untar prometheus-community/prometheus-adapter
```

Re-render Loki salt state using:
```
./charts/render.py prometheus-adapter --namespace metalk8s-monitoring \
  charts/prometheus-adapter.yaml  charts/prometheus-adapter/ \
  > salt/metalk8s/addons/prometheus-adapter/deployed/chart.sls
```

bert-e commented:
# Hello teddyandrieux,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @chengyanjin

* @lucieleonard

* @jbwatenbergscality

* @alexandre-allard-scality

* @gdemonet

* @teddyandrieux

* @nicolast

* @thomasdanan

alexandre-allard commented on salt/metalk8s/addons/prometheus-adapter/deployed/init.sls:
```diff
@@ -1,3 +1,31 @@
   - metalk8s.addons.prometheus-operator.deployed.namespace
   - .chart
+
+{#- Due to a change of prometheus-adapter deployment labelSelector in 123.0, which is immutable field
+    Manually delete the prometheus-adapter deployment object if it's an old one
+    NOTE: This logic can be removed in `development/124.0` #}
+{%- set pa_deploy = salt.metalk8s_kubernetes.get_object(
+    kind="Deployment",
+    apiVersion="apps/v1",
+    name="prometheus-adapter",
+    namespace="metalk8s-monitoring"
+) %}
+{%- if pa_deploy and salt.pkg.version_cmp(
+        pa_deploy["metadata"]["labels"]["metalk8s.scality.com/version"],
+        "123.0.0"
+    ) == -1 %}
+
+Delete the old prometheus-adapter deployment:
+  metalk8s_kubernetes.object_absent:
+    - apiVersion: apps/v1
+    - kind: Deployment
+    - name: prometheus-adapter
+    - namespace: metalk8s-monitoring
+    - wait:
+        attempts: 30
+        sleep: 10
+    - require_in:
+      - sls: metalk8s.addons.prometheus-adapter.deployed.chart
+
+{%- endif %}
```
I think the change here and for downgrade should be in a dedicated commit so we can easily do a git revert in 124.0.
Moreover, I was thinking we should create an issue to start tracking this kind of things, so when we create the next branch, we just have to look at the list of things to remove instead of searching/grepping the code and potentially miss some stuff.

TeddyAndrieux replied:
You're right, it's kind of part of the "bump" in a sense but agree that for revert it makes sense to move it to another commit, fair I will do it

TeddyAndrieux committed:
salt: Delete the prometheus-adapter before upgrade and downgrade

Since the label selector for the prometheus-adapter deployment changed
in the version we embed compare to 2.11 we need to delete this
deployment before upgrading, downgrading it

This commit should be reverted in `development/124.0`

TeddyAndrieux commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/123.0`



The following branches will **NOT** be impacted:

* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.11`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/123.0`



The following branches have **NOT** changed:

* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.11`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



Please check the status of the associated issue None.

Goodbye teddyandrieux.
